fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/Udf.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * UDF.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2024 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.metadata;
+
+import com.apple.foundationdb.record.PlanHashable;
+import com.apple.foundationdb.record.PlanSerializationContext;
+import com.apple.foundationdb.record.RecordMetaDataProto;
+import com.apple.foundationdb.record.query.plan.cascades.values.Value;
+import com.apple.foundationdb.record.query.plan.serialization.DefaultPlanSerializationRegistry;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Defines a User-defined-function.
+ */
+public class Udf {
+    @Nonnull private final String functionName;
+    @Nonnull
+    private final Value functionValue;
+
+    public Udf(@Nonnull String functionName, @Nonnull Value functionValue) {
+        this.functionName = functionName;
+        this.functionValue = functionValue;
+    }
+
+    @Nonnull
+    public String getFunctionName() {
+        return functionName;
+    }
+
+    @Nonnull
+    public Value getValue() {
+        return functionValue;
+    }
+
+    @Nonnull
+    public RecordMetaDataProto.Udf toProto() {
+        PlanSerializationContext serializationContext = new PlanSerializationContext(DefaultPlanSerializationRegistry.INSTANCE,
+                PlanHashable.CURRENT_FOR_CONTINUATION);
+        return RecordMetaDataProto.Udf.newBuilder()
+                .setFunctionName(functionName)
+                .setFunctionValue(functionValue.toValueProto(serializationContext))
+                .build();
+    }
```

Comment:
Static `fromProto(...)` missing here. See comment in `RecordMetadataBuilder`.

Reply:
done